src/tools/rbd/action/Group.cc

```diff
@@ -721,38 +727,127 @@ int execute_group_snap_list(const po::variables_map &vm,
+    if (f) {
+      f->open_object_section("group_snap");
+      f->dump_string("id", snap.id);
+      f->dump_string("snapshot", snap.name);
+      f->dump_string("state", state_string);
+      f->close_section();
     } else {
-      state_string = "ok";
+      t << snap.id << snap.name << state_string << TextTable::endrow;
     }
+  }
+
+  if (f) {
+    f->close_section();
+    f->flush(std::cout);
+  } else if (snaps.size()) {
+    std::cout << t;
+  }
+  return 0;
+}
+
+int execute_group_snap_info(const po::variables_map &vm,
+                            const std::vector<std::string> &ceph_global_args) {
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string namespace_name;
+  std::string group_name;
+  std::string group_snap_name;
+
+  int r = utils::get_pool_generic_snapshot_names(
+    vm, at::ARGUMENT_MODIFIER_NONE, &arg_index, at::POOL_NAME, &pool_name,
+    &namespace_name, GROUP_NAME, "group", &group_name, &group_snap_name, true,
+    utils::SNAPSHOT_PRESENCE_REQUIRED, utils::SPEC_VALIDATION_FULL);
+  if (r < 0) {
+    return r;
+  }
+
+  at::Format::Formatter formatter;
+  r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+  Formatter *f = formatter.get();
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  r = utils::init(pool_name, namespace_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  librbd::group_snap_info2_t group_snap;
+  r = rbd.group_snap_get_info(io_ctx, group_name.c_str(),
+                              group_snap_name.c_str(), &group_snap);
+  if (r < 0) {
+    std::cerr << "rbd: failed to show group snapshot: "
+              << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  std::vector<std::string> pool_names(group_snap.image_snaps.size());
+  for (size_t i = 0; i < group_snap.image_snaps.size(); i++) {
+    int64_t pool_id = group_snap.image_snaps[i].pool_id;
+    r = rados.pool_reverse_lookup(pool_id, &pool_names[i]);
     if (r < 0) {
+      std::cerr << "rbd: error looking up pool name for pool_id=" << pool_id
+                << ": " << cpp_strerror(r) << std::endl;
       return r;
     }
+  }
+
+  auto state_string = get_group_snap_state_name(group_snap.state);
+
+  if (f) {
+    f->open_object_section("group_snapshot");
+    f->dump_string("id", group_snap.id);
+    f->dump_string("name", group_snap.name);
+    f->dump_string("state", state_string);
+    f->dump_string("image_snap_name", group_snap.image_snap_name);
+    f->open_array_section("images");
+  } else {
+    std::cout << "rbd group snapshot '" << group_snap.name << "':\n"
+              << "\tid: " << group_snap.id << std::endl
+              << "\tstate: " << state_string << std::endl
+              << "\timage snap: " << group_snap.image_snap_name << std::endl
+              << "\timages:" << std::endl;
+  }
+
+  auto pname_itr = pool_names.begin();
+  for (const auto& image_snap : group_snap.image_snaps) {
```

Comment:
```
$ rbd group snap info data/gp@snap4
...
	images:
		rbd/img4 (snap id: 339)
		data/img1 (snap id: 12)
		data/img2 (snap id: 13)
		data/img3 (snap id: 14)
```
Since at its core this is an image listing, I think it would be nice to be consistent with the output of `rbd group image ls` command.  I see two differences:
- `rbd group image ls` sorts images based on pool ID and image name: https://github.com/ceph/ceph/blob/103cd8e78bcfe7f69647013187c053c9ccb76685/src/tools/rbd/action/Group.cc#L487-L495
- `rbd group image ls` doesn't fail if the pool name can't be produced and it's actually important.  I'll expand in a separate comment.

Do you think we could stick a similar `std::sort()` invocation here?